### PR TITLE
Fixes https://github.com/NuGet/Home/issues/1483

### PR DESCRIPTION
**Pull request by deepakaravindr**
And more known issues: https://github.com/NuGet/Home/issues/1322.
And, some unknown issues.

@yishaigalatzer @emgarten @MeniZalzman @feiling

### CODE REVIEW

File: test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class NuGetUpdateCommandTests
+    {
+        [Fact]
+        public async Task UpdateCommand_Success_References()
+        {
+            var packagesSourceDirectory = TestFilesystemUtility.CreateRandomTestFolder();
```
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs:22]: There is an opportunity to factor this code into helper methods or helper private test class. Will try to do that if time permits

**Comment by MeniZalzman** [comment]
:shipit:

**Comment by emgarten** [comment]
:rocket:

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet.PackageManagement/commit/bff6cc98060b02e91a221ce2108ba710bb922e81

**Pull request closed without merging**